src/test/java/com/google/sps/servlets/CaptionTest.java

```diff
@@ -0,0 +1,116 @@
+package com.google.sps.servlets;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class CaptionTest {
+  @Mock
+  private CaptionService captionServiceMock;
+
+  @InjectMocks
+  private Caption caption;
+
+  final private String PROPER_XML = "<?xml version=\"1.0\" encoding=\"utf-8\" ?><transcript>"
+      + "<text start=\"4.029\" dur=\"5.401\">line1</text>"
+      + "<text start=\"9.43\" dur=\"5.06\">line2</text>"
+      + "<text start=\"14.49\" dur=\"5.03\">line3</text>" + "</transcript>";
+
+  final private String PROPER_XML_CAPTION = "line1 line2 line3";
+
+  final private String BAD_XML = "<?xml version=\"1.0\" encoding=\"utf-8\" ?><transcript>"
+      + "<text start=\"4.029\" dur=\"5.401\">line1</text>";
+
+  final private String FAILED_CAPTION = null;
+
+  final private String VIDEO_ID = "sampleId";
+
+  final private String LANG_EN = "en";
+
+  final private String LANG_EN_GB = "en-GB";
+
+  @Before
+  public void init() {
+    caption = new Caption();
+    MockitoAnnotations.initMocks(this);
+  }
+
+  private InputStream streamOn(String xml) {
+    return new ByteArrayInputStream(xml.getBytes());
+  }
+
+  @Test
```

Comment:
Is the throws IOException needed. Seems smelly if all unit tests have it.

Reply:
The tests can't build if I remove it from the function signature. Is this not the case in other testing packages?

Reply:
Which method in the unit test can throw IOException? Is it the ByteArrayInputStream constructor? Or is it something else? Trying to see if there is a way to clean it off.

Reply:
It is the captionServiceMock.fetchStream() call.